src/lib/autocomplete/autocomplete-trigger.ts

```diff
@@ -343,6 +342,29 @@ export class MdAutocompleteTrigger implements ControlValueAccessor, OnDestroy {
     }
   }
 
+  /** Determines the index of the active option. */
+  private _getActiveOptionIndex(): number {
+    let optionIndex = this.autocomplete._keyManager.activeItemIndex || 0;
+
+    // If there are any option groups, we need to offset
+    // the index by the amount of groups that come before the option.
+    if (this.autocomplete.optionGroups.length) {
+      const options = this.autocomplete.options.toArray();
+      const groups = this.autocomplete.optionGroups.toArray();
+      let groupCounter = 0;
+
+      for (let i = 0; i < optionIndex + 1; i++) {
```

Comment:
Sure, but I'm not sure what that would even be called. `countGroupLabelsBeforeOption`? I would rather put it as a static method on the `MdOption` instead since having it float around as a function implies that it's a bit more reusable.